src/app-layer-quic.c

```diff
@@ -0,0 +1,57 @@
+/* Copyright (C) 2020 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+/**
+ * \file
+ *
+ * Quic Application Layer
+ *
+ */
+
+#include "suricata-common.h"
+#include "stream.h"
+#include "conf.h"
+
+#include "util-unittest.h"
+
+#include "app-layer-detect-proto.h"
+#include "app-layer-parser.h"
+
+#include "app-layer-quic.h"
+#include "rust.h"
+
+void RegisterQuicParsers(void)
+{
+    /* Only register if enabled in config. */
+    if (ConfGetNode("app-layer.protocols.quic") == NULL) {
+        return;
+    }
+
```

Comment:
Are there any magic patterns to detect this protocol on non-standard ports ?

Reply:
Not that I know of. This could be added in a future PR.